src/Mvc/Mvc.RazorPages/src/ApplicationModels/CompiledPageActionDescriptorBuilder.cs

```diff
@@ -61,6 +61,18 @@ internal static class CompiledPageActionDescriptorBuilder
             };
         }
 
+        private static IList<object> CreateEndPointMetadata(PageApplicationModel applicationModel)
+        {
+            var handlerMetatdata = applicationModel.HandlerTypeAttributes;
+            var endPointMetadata = applicationModel.EndpointMetadata;
+
+            // It is criticial to get the order in which metadata appears in endpoint metadata correct. More significant metadata
+            // must appear later in the sequence.
+            // In this case, handlerMetadata is attributes on the Page \ PageModel, and endPointMetadata is configured via conventions. and 
```

Comment:
`AuthorizePage(..)` would have added at the end of the filter collection making it the effective filter. This matches the behavior

Reply:
Did we do this before? I really don't want to make attributes on handler methods *just work* since they don't work on legacy routing and legacy routing is still supported.

Reply:
Ignore me. This says handler type.